### ISSUE TITLE
Ignore rel="stylesheet" in H59 "Using the link element and navigation tools"

### ISSUE THREAD

**Issue by Maxim-Mazurok** [w3c/wcag#2297]
I'm trying to add `<link rel="stylesheet" href="blabla.css">` in the HTML body, but rule H59 (`WCAG2AAA.Principle2.Guideline2_4.2_4_8.H59.1`) doesn't let me: "Link elements can only be located in the head section of the document."

In the [description](https://www.w3.org/TR/WCAG20-TECHS/H59#H59-description), it says:

> Several values of rel are useful:
> 
> Start
> Next 
> Prev
> Contents
> Index

So, I think there's not much sense in preventing me from putting styles in HTML, because "stylesheet" is definitely not a navigation link.

Either we can exclude "stylesheet" as the most popular type of link. Or we can include only Start, Next, etc. into checks.

Or we can go one step further and get rid of this technique completely, but I didn't do much research on this. See https://github.com/w3c/wcag/issues/1234 and https://github.com/w3c/wcag/issues/1036 by @patrickhlauke

**Comment by patrickhlauke** [comment]
Note that this error/failure you're seeing has nothing to do with H59. The error you're getting appears to be something related to squizlab's code checker and its own set of checks/rules. At a stretch, this may be a mild failure of 4.1.1 but certainly unrelated to 2.4.5

**Comment by Maxim-Mazurok** [comment]
> [body-ok](https://html.spec.whatwg.org/multipage/links.html#body-ok) is a link attribute.

Wow, that's a great find!

> Late-loading stylesheets seems like the sort of thing that could easily trip up AT.

Yes, that's exactly why I'm putting it there, so that browser will only start loading style once it loaded preceding HTML, so it renders before applying additional cosmetic non-layout styles. Not sure who is AT...

**Comment by Maxim-Mazurok** [comment]
> Note that this error/failure you're seeing has nothing to do with H59. The error you're getting appears to be something related to squizlab's code checker and its own set of checks/rules. At a stretch, this may be a mild failure of 4.1.1 but certainly unrelated to 2.4.5

I'm not using squizlab, I'm using https://github.com/pa11y/pa11y

**Comment by Maxim-Mazurok** [comment]
I think I misread the procedure on w3c website, it does say to check links related to navigation. Even though it doesn't say to check **only** links related to navigation, I still think that there's not much to fix here (apart from potentially removing this, which has 2 other open issues).
So I opened one here: https://github.com/squizlabs/HTML_CodeSniffer/issues/306
And closing this one, cheers!

**Comment by bruce-usab** [comment]
@Maxim-Mazurok  -- My apologies:  AT is assistive technology.  An example is the JAWS screen reading software, used by many MS Windows users who are blind.
